### PR TITLE
Add dotnet sample to the set-up-subscriptions repository

### PR DESCRIPTION
**Pull request by stefan-stripe**
r? @ctrudeau-stripe @adreyfus-stripe 

Description: Adding dotnet sample for setting up subscriptions.

- Created using `dotnet new mvc`
- Changed default webroot to `../../client`
- Using `SnakeCaseNamingStrategy` for JSON serialization settings to work with the sample client code.
- Loading settings from `appsettings.json`, will look into making it similar to the other samples and potentially using the .env file.

Let me know what you think, next steps for me are:

- Add a sample that does a bit more than creating customers/subscriptions.

### CODE REVIEW

File: server/dotnet/Program.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                    webBuilder.UseWebRoot("../../client/");
```
Comment (ctrudeau-stripe) [inline server/dotnet/Program.cs]: Can we add this as an environment variable? Not sure how that works in .NET

File: server/dotnet/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "sample": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:5001;http://localhost:5000",
```
Comment (ctrudeau-stripe) [inline server/dotnet/Properties/launchSettings.json]: Can we make this localhost:4242?

**Review by ctrudeau-stripe — APPROVED** [review]
LGTM, want to change to localhost:4242 to match the others.

File: server/dotnet/Program.cs (hunk 1 above)
Comment (adreyfus-stripe) [reply server/dotnet/Program.cs]: +1, our .env.example has it set as `STATIC_DIR`

Comment (adreyfus-stripe) [reply server/dotnet/Program.cs]: Let's also add a comment that this is so we can reuse the same client code across different server languages and not a requirement to get it to work with Stripe

File: server/dotnet/Properties/launchSettings.json (hunk 2 above)
Comment (stefan-stripe) [reply server/dotnet/Properties/launchSettings.json]: You got it!

File: server/dotnet/Program.cs (hunk 1 above)
Comment (stefan-stripe) [reply server/dotnet/Program.cs]: Added reading the web root path from `STATIC_DIR` and additional comment to explain this is only for sample purposes

**Comment by thorsten-stripe** [comment]
Sorry for the lurk, and maybe I'm just blind, but does this implement webhook handling?

**Comment by stefan-stripe** [comment]
> Sorry for the lurk, and maybe I'm just blind, but does this implement webhook handling?

No, it doesn't but I'll go ahead and add!

**Pull request merged**